src/shm/pool/slot.rs

```diff
@@ -302,7 +302,9 @@ impl SlotPool {
         stride: i32,
         format: wl_shm::Format,
     ) -> Result<Buffer, CreateBufferError> {
-        let offset = slot.inner.offset as i32;
+        let mut offset = slot.inner.offset as i32;
+        offset += offset % 64;
+        offset -= offset % 64;
```

Comment:
This change just breaks things; it does not actually align anything.  The current alignment code is on line 259.

Reply:
I'm still not sure if I broke something.

Reply:
The initial `x += x % y;` statement does not help with aligning `x`; that's what breaks things.  The second statement will correctly align values, but does so by rounding down - and at this point, that just ends up overlapping the prior buffer.